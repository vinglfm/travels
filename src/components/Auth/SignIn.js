import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Validaton from '../../_common/Validation';
import actions from '../../_actions';
import { connect } from 'react-redux';
import styles from './AuthModal.css';

export class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };

    this.validation = new Validaton();
    this.signIn = this.signIn.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  signIn(event) {
    this.props.signIn(this.state);
    this.props.onClose();
    event.preventDefault();
  }

  handleChange(elem) {
    this.setState({ [elem.target.name]: elem.target.value });
  }

  render() {
    const emailError = this.validation.validateEmail(this.state);
    const passwordError = this.validation.validatePassword(this.state);
    const emailErrorMessage = emailError[0] ? emailError[0].message : '';
    const passwordErrorMessage = passwordError[0] ? passwordError[0].message : '';
    return (
      <div>
        <div className={styles.modal__header}>
          <a onClick={this.props.onBack} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
          <span>Sign In</span>
        </div>
        <hr className={styles.modal__divider} />
        <form className={styles.modal__form} name='SignInForm' onSubmit={this.signIn} noValidate>
          <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
          <input className={emailError === true || !this.state.email ? styles.modal__form__input : styles.modal__form__input__error} name='email' type='email' placeholder='Email' onChange={this.handleChange} />
          <div className={styles.modal__form__error}>
            <span className={`${styles.modal__form__error__message} ${!this.state.email ? styles.modal__form__error__message__hidden : ''}`}>{emailErrorMessage}</span>
          </div>
          <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
          <input className={passwordError === true || !this.state.password ? styles.modal__form__input : styles.modal__form__input__error} name='password' type='password' placeholder='Password' onChange={this.handleChange} />
          <div className={styles.modal__form__error}>
            <span className={`${styles.modal__form__error__message} ${!this.state.password ? styles.modal__form__error__message__hidden : ''}`}>{passwordErrorMessage}</span>
          </div>
          <div className={styles.modal__form__group}>
            <input className={styles.modal__form__group__submit} type='submit' disabled={emailError !== true || passwordError !== true} value='Sign In' />
          </div>
        </form>
      </div>
    );
  }
}

SignIn.propTypes = {
  onBack: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  signIn: actions.signIn
};

export default connect(null, mapDispatchToProps)(SignIn);