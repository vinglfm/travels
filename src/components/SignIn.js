import React, {Component} from 'react';
import FacebookLogin from 'react-facebook-login';
import PropTypes from 'prop-types';
import styles from './SignInModal.css';

export default class SignIn extends Component {

    facebookResponse (resp) {
        console.log(resp);
    };

    render() {
        return (
            <div>
            <FacebookLogin
            appId='1817436878556969'
            autoLoad={false}
            fields='name,email'
            cssClass={styles.fb__button}
            callback={this.facebookResponse}
            textButton='Facebook'
            />
            <div className={styles.lines}>
                <span>or</span>
            </div>
            <div className={styles.log__in}><a onClick={this.props.logInComponent}>Log In</a></div>
            <div className={styles.sign__up}>First time travel? <a onClick={this.props.signUpComponent}>Sign Up</a></div>
        
            </div>
        );
    }
}

SignIn.propTypes = {
    logInComponent: PropTypes.func.isRequired,
    signUpComponent: PropTypes.func.isRequired,
};