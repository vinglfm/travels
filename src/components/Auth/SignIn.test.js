import enzyme, {shallow} from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactSixteenAdapter() });

import React from 'react';
import {SignIn} from './SignIn';

it('renders SignIn', () => {
  const snapshot = shallow(<SignIn onSignIn={()=>{}} />);
  expect(snapshot).toMatchSnapshot();
});

it('renders LogIn component when state is chagned to LogIn', () => {
  const snapshot = shallow(<SignIn onSignIn={()=>{}} />);
  snapshot.instance().renderComponent('LogIn');
  snapshot.update();
  expect(snapshot).toMatchSnapshot();
});

it('renders SignUp component when state is changed to SignUp', () => {
  const snapshot = shallow(<SignIn onSignIn={()=>{}} />);
  snapshot.instance().renderComponent('SignUp');
  snapshot.update();
  expect(snapshot).toMatchSnapshot();
});
