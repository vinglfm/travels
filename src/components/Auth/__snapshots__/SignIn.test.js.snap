// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LogIn component when state is chagned to LogIn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    onSignIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <LogIn
        onBack={[Function]}
        onLogIn={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onBack": [Function],
        "onLogIn": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <LogIn
          onBack={[Function]}
          onLogIn={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onBack": [Function],
          "onLogIn": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders SignIn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    onSignIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <t
                appId="1817436878556969"
                autoLoad={false}
                callback={[Function]}
                cssClass="kep-login-facebook"
                fields="name,email"
                size="metro"
                tag="button"
                textButton="Facebook"
                typeButton="button"
        />
        <div
                className={undefined}
        >
                <span>
                        or
                </span>
        </div>
        <div
                className={undefined}
        >
                <a
                        onClick={[Function]}
                >
                        Log In
                </a>
        </div>
        <div
                className={undefined}
        >
                First time travel? 
                <a
                        onClick={[Function]}
                >
                        Sign Up
                </a>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <t
            appId="1817436878556969"
            autoLoad={false}
            callback={[Function]}
            cssClass="kep-login-facebook"
            fields="name,email"
            size="metro"
            tag="button"
            textButton="Facebook"
            typeButton="button"
/>,
          <div
            className={undefined}
>
            <span>
                        or
            </span>
</div>,
          <div
            className={undefined}
>
            <a
                        onClick={[Function]}
            >
                        Log In
            </a>
</div>,
          <div
            className={undefined}
>
            First time travel? 
            <a
                        onClick={[Function]}
            >
                        Sign Up
            </a>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appId": "1817436878556969",
            "autoLoad": false,
            "callback": [Function],
            "cssClass": "kep-login-facebook",
            "fields": "name,email",
            "size": "metro",
            "tag": "button",
            "textButton": "Facebook",
            "typeButton": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              or
</span>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "or",
            },
            "ref": null,
            "rendered": "or",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
>
              Log In
</a>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Log In",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "First time travel? ",
              <a
                onClick={[Function]}
>
                Sign Up
</a>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            "First time travel? ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <t
                    appId="1817436878556969"
                    autoLoad={false}
                    callback={[Function]}
                    cssClass="kep-login-facebook"
                    fields="name,email"
                    size="metro"
                    tag="button"
                    textButton="Facebook"
                    typeButton="button"
          />
          <div
                    className={undefined}
          >
                    <span>
                              or
                    </span>
          </div>
          <div
                    className={undefined}
          >
                    <a
                              onClick={[Function]}
                    >
                              Log In
                    </a>
          </div>
          <div
                    className={undefined}
          >
                    First time travel? 
                    <a
                              onClick={[Function]}
                    >
                              Sign Up
                    </a>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <t
              appId="1817436878556969"
              autoLoad={false}
              callback={[Function]}
              cssClass="kep-login-facebook"
              fields="name,email"
              size="metro"
              tag="button"
              textButton="Facebook"
              typeButton="button"
/>,
            <div
              className={undefined}
>
              <span>
                            or
              </span>
</div>,
            <div
              className={undefined}
>
              <a
                            onClick={[Function]}
              >
                            Log In
              </a>
</div>,
            <div
              className={undefined}
>
              First time travel? 
              <a
                            onClick={[Function]}
              >
                            Sign Up
              </a>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appId": "1817436878556969",
              "autoLoad": false,
              "callback": [Function],
              "cssClass": "kep-login-facebook",
              "fields": "name,email",
              "size": "metro",
              "tag": "button",
              "textButton": "Facebook",
              "typeButton": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                or
</span>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "or",
              },
              "ref": null,
              "rendered": "or",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
>
                Log In
</a>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log In",
              "type": "a",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "First time travel? ",
                <a
                  onClick={[Function]}
>
                  Sign Up
</a>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              "First time travel? ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders SignUp component when state is changed to SignUp 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    onSignIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <SignUp
        onBack={[Function]}
        onSignUp={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onBack": [Function],
        "onSignUp": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <SignUp
          onBack={[Function]}
          onSignUp={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onBack": [Function],
          "onSignUp": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
